### PR TITLE
Add KDP - Generate Existing Resources

### PR DESCRIPTION
**Pull request by prateekpandey14**
Signed-off-by: prateekpandey14 <prateek.pandey@nirmata.com>

Converted the [google docs](https://docs.google.com/document/d/1KHf19cV5o8fWWC78OBl3H1-OETRuzHciEEBTfRFfMUU/edit#heading=h.w8ahammwm4ug)  to a KDP proposal template

### CODE REVIEW

**Review by JimBugwadia — APPROVED** [review]

**Comment by chipzoller** [comment]
Same with this. The feature is implemented in 1.7.0 yet the KDP was never formally accepted.

**Review by vyankyGH — APPROVED** [review]

**Comment by chipzoller** [comment]
Awaiting final comment resolution from Shutting.

File: proposals/generate_existing_resources.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+Author: [Prateek Pandey](https://github.com/prateekpandey14)
+
+  * [Overview](#overview)
+  * [Proposal](#proposal)
+  * [Implementation](#implementation)
+  * [Example](#example)
+
+## Overview
+
+Generate-Policy Controller runs as an admission controller in Kubernetes watches
```
[the remark this answers is not in the document]
Comment (prateekpandey14) [reply proposals/generate_existing_resources.md:13]: Right , fixed.

File: proposals/generate_existing_resources.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        kinds:
+        - v1/ServiceAccount
+```
+
+Creating a ClusterPolicy with a rule containing a match statement which matches on kind as well as the label you have set on the resource, but this feels very counter-intuitive and cumbersome, hence the new proposal to make this easy to use.
+
+
+## Proposal
+
+- Whenever trigger resource is created or updated, kyverno will  see the policy update, if it finds that policy has the `genExisting:true` field set , the new generate background scan handler will scan through all the namespaces with matching Kind and then trigger the current generate controller workflow to handle the generate request operations.
```
Comment (realshuting) [inline proposals/generate_existing_resources.md:67]: One last generate request here ^

Comment (prateekpandey14) [reply proposals/generate_existing_resources.md]: done

**Review by realshuting — APPROVED** [review]

**Pull request merged**